httpgrpc/server.go

```diff
@@ -180,9 +176,11 @@ func handleMethod(svr interface{}, serviceName string, desc *grpc.MethodDesc, un
 			return
 		}
 
-		codec := encoding.GetCodec(grpcproto.Name)
 		dec := func(msg interface{}) error {
-			return codec.Unmarshal(req, msg)
+			if err := codec.Unmarshal(req, msg); err != nil {
+				return status.Error(codes.InvalidArgument, err.Error())
```

Comment:
FWIW, we probably should have landed this change ages ago; the current behavior is internal error.